server/src/main/java/org/elasticsearch/search/aggregations/metrics/AbstractPercentilesAggregationBuilder.java

```diff
@@ -126,6 +129,7 @@
         this.values = sortedValues;
         this.percentilesConfig = percentilesConfig;
         this.valuesField = valuesField;
+        this.metricNames = Arrays.stream(values).mapToObj(String::valueOf).collect(Collectors.toSet());
```

Comment:
I think most paths won't actually use the set. Maybe we'd be better off building it in the method implementation?

Reply:
fair enough, move it to the method implementation.